script.js

```diff
@@ -1,4 +1,339 @@
+    return true;
+  } else {
+    return false;
+  }
+};
+var checkDealerForBlackJack = function () {
+  var dealerTotalValue = 0;
+  for (var i = 0; i < dealerCards.length; i++) {
+    var cardValue = dealerCards[i].rank;
+    if (cardValue == 13 || cardValue == 12 || cardValue == 11) {
+      cardValue = 10;
+    } else if (cardValue === 1) {
+      if (dealerTotalValue + 11 <= 21) {
+        cardValue = 11;
+      } else {
+        cardValue = 1;
+      }
+    }
+    dealerTotalValue += Number(cardValue);
+  }
+  if (dealerTotalValue == 21) {
+    return true;
+  } else {
+    return false;
+  }
+};
+function calculatePlayerHandValue(playerCards) {
+  var totalValue = 0;
+  var aceCount = 0;
+  for (var i = 0; i < playerCards.length; i++) {
+    var cardValue = playerCards[i].rank;
+    if (cardValue == 13 || cardValue == 12 || cardValue == 11) {
+      cardValue = 10;
+    } else if (cardValue === 1) {
+      cardValue = 11;
+      aceCount++;
+    }
+    totalValue += Number(cardValue);
+  }
+  while (totalValue > 21 && aceCount > 0) {
+    totalValue -= 10;
+    aceCount--;
+  }
+  return totalValue;
+}
+function calculateDealerHandValue(dealerCards) {
+  var totalValue = 0;
+  var aceCount = 0;
+  for (var i = 0; i < dealerCards.length; i++) {
+    var cardValue = dealerCards[i].rank;
+    if (cardValue == 13 || cardValue == 12 || cardValue == 11) {
+      cardValue = 10;
+    } else if (cardValue === 1) {
+      cardValue = 11;
+      aceCount++;
+    }
+    totalValue += Number(cardValue);
+  }
+  while (totalValue > 21 && aceCount > 0) {
+    totalValue -= 10;
+    aceCount--;
+  }
+  return totalValue;
+}
+
+//edit this for display blackjack
+var displayHands = function (playerCards, dealerCards) {
+  console.log("Player's hand:");
+  var playerMessage = "Player's hand: <br>";
+  var dealerMessage = "Dealer's hand: <br>";
+  for (let i = 0; i < playerCards.length; i++) {
+    playerMessage =
+      playerMessage +
+      " - " +
+      playerCards[i].name +
+      " of " +
+      playerCards[i].suit +
+      "<br>";
+    console.log(`${playerCards[i].rank} of ${playerCards[i].suit}`);
+  }
+  console.log("Dealers's hand:");
+  for (let i = 0; i < dealerCards.length; i++) {
+    dealerMessage =
+      dealerMessage +
+      " - " +
+      dealerCards[i].name +
+      " of " +
+      dealerCards[i].suit +
+      "<br>";
+    console.log(`${dealerCards[i].rank} of ${dealerCards[i].suit}`);
+  }
+  //future use
+  /*console.log("\nDealer's hand:");
+  console.log(`${dealerCards[0].rank} of ${dealerCards[0].suit}`);
+  console.log("Hidden card");*/
+  return playerMessage + "<br>" + dealerMessage;
+};
+var displayHandsValue = function (playerHandTotalValue, dealerHandTotalValue) {
+  var totalHandValueMessage =
+    "<br> Player total hand value: " +
+    playerHandTotalValue +
+    "<br> Dealer total hand value: " +
+    dealerHandTotalValue;
+  return totalHandValueMessage;
+};
 var main = function (input) {
```

Comment:
The main function is pretty long, you can probably abstract out quite a few blocks into their own functions